source/common/router/router.cc

```diff
@@ -353,11 +353,14 @@ Http::FilterHeadersStatus Filter::decodeHeaders(Http::HeaderMap& headers, bool e
     return Http::FilterHeadersStatus::StopIteration;
   }
 
-  // Fetch a connection pool for the upstream cluster.
-  Http::ConnectionPool::Instance* conn_pool = getConnPool();
-  if (!conn_pool) {
-    sendNoHealthyUpstreamResponse();
-    return Http::FilterHeadersStatus::StopIteration;
+  Http::ConnectionPool::Instance* conn_pool = nullptr;
```

Comment:
Can you explain this change more? Related to my comment below.

Reply:
Removed. Sorry!

Reply:
more to revert here?

Reply:
Fixed. Thanks!